cli/bpmetadata/cmd.go

```diff
@@ -148,6 +157,21 @@ func generateMetadataForBpPath(bpPath string) error {
 		return fmt.Errorf("error creating metadata for blueprint at path: %s. Details: %w", bpPath, err)
 	}
 
+	// If the flag is set, extract output types
+	if mdFlags.genOutputType {
+		outputTypes, err := extractOutputTypesFromState(bpPath)
+		if err != nil {
+			return fmt.Errorf("error extracting output types for blueprint at path: %s. Details: %w", bpPath, err)
+		}
+
+		// Update the bpMetadata with the extracted output types
+		for i, output := range bpMetaObj.Spec.Interfaces.Outputs {
+			if outputType, ok := outputTypes[output.Name]; ok {
+				bpMetaObj.Spec.Interfaces.Outputs[i].Type = outputType
+			}
+		}
+	}
+
```

Comment:
Could we keep this in getBlueprintInterfaces which is where the other input/output related logic exists? This will also allow us to test this https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/blob/17331bbd4633f0f56fc07b67700542f8df43521f/cli/bpmetadata/tfconfig_test.go

Reply:
Make sense to me. Made the changes and added tests in [this commit](https://github.com/GoogleCloudPlatform/cloud-foundation-toolkit/pull/2602/commits/7dbdcb144b42d9eaae23296247605261a1850e07)